curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64fb1c4dc0feb219149a7c7d.md

```diff
@@ -11,6 +11,10 @@ If the user attempts to edit a task but decides not to make any changes before c
 
 Inside the `closeTaskFormBtn` event listener, use `const` to create another variable named `formInputValuesUpdated`. Check if the user made changes while trying to edit a task by verifying that the `titleInput` value **is not equal to** `currentTask.title`, the `dateInput` value **is not equal to** `currentTask.date`, and the `descriptionInput` value **is not equal to** `currentTask.description`.
 
+If you try to add a new task, edit that task, and then click on the `"Add New task"` button, you will notice a bug.
+
+The form button will display the incorrect text of `"Update Task"` instead of `"Add Task"`. To fix this, you will need to update your `addOrUpdateTask` function to update the `innerText` of the `addOrUpdateTaskBtn` to `"Add Task"`.
+
```

Comment:
You have misunderstood the instructions from the issue.

You do not need to add the description to the existing step 46.

You need to create a new step 46.
Please remove this here.

Then read the docs on how to insert a new step 46 liked mentioned in the issue.
https://contribute.freecodecamp.org/#/how-to-work-on-practice-projects?id=how-to-work-on-practice-projects

To insert a new step you can either use the challenge editor or run the following command inside the todo app project directory. Everything is laid out in the docs.

```
pnpm run insert-step 46
```


```suggestion
```

Reply:
You also need to make sure there are tests for this too

Reply:
500 Internal Server Error
I am getting this while creating new step 46.

Reply:
Have you tried the `pnpm` command instead?

Stop the challenge editor and instead open up a new tab in your terminal.

Then `cd` to the project directory

```
cd curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/
```
Then run the insert step command

```
pnpm run insert-step 46
```

Reply:
Yes, I tried the ```pnpm``` command and it is still showing an error.

Error: ENOENT: no such file or directory, open 'E:\freeCodeCamp\curriculum\challenges\_meta\E:\freeCodeCamp\curriculum\challenges\english\15-javascript-algorithms-and-data-structures-22\learn-localstorage-by-building-a-todo-app\meta.json'

Reply:
I just pushed an update to your branch to include the new step.

You should now see a new step 46

You will still have to add in the directions and tests